### PR TITLE
Remove some cross project compile includes on cs files

### PR DESCRIPTION
**Pull request by krwq**
part of #5070 fix

Note file renames

Here is list of stuff which is left
- `D:\cli\src\dotnet-archive\dotnet-archive.csproj`
  - `..\dotnet\CommandLine\*.cs;..\dotnet\CommonLocalizableStrings.cs;`
- `D:\cli\TestAssets\TestPackages\dotnet-dependency-tool-invoker\dotnet-dependency-tool-invoker.csproj`
  - `..\..\..\src\dotnet\CommandLine\*.cs;..\..\..\src\dotnet\CommonLocalizableStrings.cs;`
- `D:\cli\tools\Archiver\Archiver.csproj`
  - `..\..\src\dotnet-archive\*.cs;..\..\src\dotnet\CommandLine\*.cs;..\..\src\dotnet\CommonLocalizableStrings.cs;`

They will be addressed separately as they require a little more work which is not fully clear yet

cc: @jgoshi @livarcocc @piotrpMSFT @jonsequitur

### CODE REVIEW

**Comment by krwq** [comment]
last commit is a fail - i accidentally removed my own change and reordered stuff, fixing soon

File: src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal static class DotnetFiles
         /// </summary>
         public static string VersionFile => Path.GetFullPath(Path.Combine(SdkRootFolder, ".version"));
 
-        internal static DotnetVersionFile VersionFileObject
+        public static DotnetVersionFile VersionFileObject
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs]: Can you leave this internal and do a InternalsVisibleTo?

File: src/Microsoft.DotNet.Cli.Utils/DotnetVersionFile.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.DotNet.Cli.Utils
 {
-    internal class DotnetVersionFile
+    public class DotnetVersionFile
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/DotnetVersionFile.cs]: Ditto.

File: src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 namespace Microsoft.DotNet.Cli
 {
-    internal static class DotnetFiles
+    public static class DotnetFiles
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs]: Leave internal.

File: src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs (hunk 1 above)
Comment (krwq) [inline src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs]: @livarcocc I believe this is useful outside of this dll too - why InternalsVisibleTo?

Comment (krwq) [inline src/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs]: talked offline - changing to internal

**Comment by krwq** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by krwq** [comment]
@srivatsn can I get approval for this? this is decoupling some components which previously slowed us down on last minute fixes

**Comment by piotrpMSFT** [comment]
nice!

**Pull request merged**